src/external_api.py

```diff
@@ -0,0 +1,114 @@
+import logging
+import os
+from datetime import datetime, timedelta
+from typing import Dict, Union
+
+import requests
+from dotenv import load_dotenv
+
+from logging_config import setup_logging
+
+setup_logging()
+logger = logging.getLogger("my_log")
+
+load_dotenv()
+CURRENCY_API_KEY = os.getenv("CURRENCY_API_KEY")
+STOCK_API_KEY = os.getenv("STOCK_API_KEY")
+
+
+def get_currency_rate(currency: str, amount: int = 1) -> Dict[str, Union[str, float]]:
+    """
+    Функция, принимающая на вход валюту для конвертации, сумму (по умолчанию 1) и возвращающая словарь в формате:
+    {"currency": "валюта",
+     "rate": стоимость в рублях}
+    """
+    logger.info("Определяем url и др. необходимые объекты")
+
+    url = f"https://api.apilayer.com/exchangerates_data/convert?to=RUB&from={currency}&amount={amount}"
```

Comment:
Параметры корректнее передавать отдельным аргументом
```suggestion
    url = 'https://api.apilayer.com/exchangerates_data/convert'
    params = {'to': 'RUB', 'from': currency, 'amount': amount}
    headers = {'apikey': CURRENCY_API_KEY}
    response = requests.get(url, params=params, headers=headers}
```